app/docs/dev.md

```diff
@@ -1,29 +1,207 @@
-# YourFunctionName
-*Brief description of what this evaluation function does, from the developer perspective*
+# buckinghamPiTheorem
+
+Checks that the set of quantities in the response matches the set of quantities in the sense given by the Buckingham Pi theorem.
+
+There are three different ways of supplying this function with the necessary information.
+1. In the answer, provide an example set of groups as a comma separated list. When used this way the function assumes that the given list is correct and contains at least the minimum number of groups.
+2. In the `quantities` parameter, supply a list of what the dimensions for each quantity is and set answer to `-`. The function will then compute a list of sufficiently many independent dimensionless quantities and compare to the response.
+3. In the `quantities` parameter, supply a list of what the dimensions for each quantity is and in the answer, supply a list of groups as in the first option. The function will then check that the supplied answer is dimensionless and has a sufficient number of independent groups before comparing it to the response.
+
+Note that in lists of groups the items should ideally be written in the form $q_1^{c_1} \cdot q_2^{c_2} \cdots q_n^{c_n}$ where $q_1, q_2 \ldots q_n$ are quantities and $c_1, c_2 \ldots c_n$ are integers, but the function can also handle item that are sums with terms written in the form $a \cdot q_1^{c_1} \cdot q_2^{c_2} \cdots q_n^{c_n}$ where $q_1, q_2 \ldots q_n$ are quantities, $c_1, c_2 \ldots c_n$ rational numbers and $a$ a constant. If the total number of groups is less than required the set of groups is considered invalid, even if there is a sufficient number of terms with independent power products in the response.
+
+The core part of the function is an algorithm that takes two sets of groups and detmines if they generate the same nullspace. If one set of groups is valid and both groups generate the same nullspace then the second set of groups is also valid. It is assumed that the task author either supplies a valid set (or details on the dimensions of the quantities involved such that a valid set can be generated) that can used to check the validity of response. For a more detailed discussion of the algorithm see: [Automated feedback on student attempts to produce a set of dimensionless power products from a list of quantities that describe a physical problem](https://doi.org/10.6084/m9.figshare.24850131), the evaluation implements the algorithm described there (see `determine_validity` in `evaluation.py`) together with some modifications (see below).
```

Comment:
typo here - suggest spell check the whole document

Reply:
Sentence rephrased